tests/test_buddycheck.py

```diff
@@ -0,0 +1,163 @@
+import unittest
+from unittest.mock import patch
+import pandas as pd
+import tempfile
+import os
+from cautiousrobot import BuddyCheck
+from cautiousrobot.utils import process_csv
+
+class TestBuddyCheck(unittest.TestCase):
+    def setUp(self):
+        self.buddy_check = BuddyCheck()
+
+        self.img_source_file = tempfile.NamedTemporaryFile(delete=False, mode='w')
+        self.checksum_source_file = tempfile.NamedTemporaryFile(delete=False, mode='w')
+
+        self.img_source_file.write("""filename,checksum
+                                    image1.jpg,abc123
+                                    image2.jpg,def456
+                                    image3.jpg,ghi789
+                                    """)
+        self.img_source_file.close()
+
+        self.checksum_source_file.write("""filename,md5
+                                    image1.jpg,abc123
+                                    image2.jpg,def456
+                                    image3.jpg,ghi789
+                                    """)
+        self.checksum_source_file.close()
+
+    def tearDown(self):
+        os.remove(self.img_source_file.name)
+        os.remove(self.checksum_source_file.name)
+
+    def test_initialization(self):
+        self.assertEqual(self.buddy_check.buddy_id, None)
+        self.assertEqual(self.buddy_check.buddy_col, 'md5')
+
+    def test_merge_on_checksum(self):
+        source_df = pd.read_csv(self.img_source_file.name)
+        checksum_df = pd.read_csv(self.checksum_source_file.name).drop(columns=['filename'])
+
+        merged_df = self.buddy_check.merge_on_checksum(source_df, checksum_df, 'checksum')
+        expected_df = pd.DataFrame({
+            'filename': ['image1.jpg', 'image2.jpg', 'image3.jpg'],
+            'checksum': ['abc123', 'def456', 'ghi789'],
+            'md5': ['abc123', 'def456', 'ghi789']
+        })
+        pd.testing.assert_frame_equal(merged_df, expected_df)
+
+    def test_merge_on_filename_checksum(self):
+        source_df = pd.read_csv(self.img_source_file.name)
+        checksum_df = pd.read_csv(self.checksum_source_file.name)
+        self.buddy_check.buddy_id = 'filename'
+        merged_df = self.buddy_check.merge_on_filename_checksum(source_df, checksum_df, 'filename', 'checksum')
+        expected_df = pd.DataFrame({
+            'filename': ['image1.jpg', 'image2.jpg', 'image3.jpg'],
+            'checksum': ['abc123', 'def456', 'ghi789'],
+            'md5': ['abc123', 'def456', 'ghi789']
+        })
+        pd.testing.assert_frame_equal(merged_df, expected_df)
+
+    def test_check_alignment_all_matching(self):
```

Comment:
`self.buddy_check.buddy_id = 'filename'` must be set again or you can set another `BuddyCheck` instance with the `buddy_id` defined for those using it (potentially the more efficient method). I'm guessing this will resolve the 6 BuddyCheck tests that are failing.

Reply:
Thank you! I have set up another BuddyCheck instance with the buddy_id defined. Now, there are no errors, just a few failed test cases. I will investigate further.

Reply:
Hello @egrace479. I made these changes: 1. add another BuddyCheck instance with buddy_id initialized. 2. removing the leading whitespace with self.image_source_file and self.check_source_file in setUp. 3. update missing_imgs to retain its original index. Now all the test cases with test_buddycheck run clear.